setup.py

```diff
@@ -1,10 +1,5 @@
-import os
 from setuptools import setup
-from setuptools.config import read_configuration
+from kaitaistruct import __version__
 
-this_dir = os.path.dirname(__file__)
-cfg = read_configuration(os.path.join(this_dir, 'setup.cfg'))
-cfg["options"].update(cfg["metadata"])
-cfg = cfg["options"]
-
-setup(**cfg)
+if __name__ == "__main__":
+    setup()
```

Comment:
Not sure if this guard is necessary - there's no legitimate use case for importing setup.py as a module. If anything, I would throw an exception if `__name__ != "__main__"`.

Reply:
The guard is not really necessary, but it is the right thing to add it, so I have done it.

>If anything, I would throw an exception if __name__ != "__main__".

IMHO we shouldn't. We cannot predict all the ways third party code can use our code and we shouldn't counteract it.

Reply:
> We cannot predict all the ways third party code can use our code

Actually in this case we can - all previous versions of `kaitaistruct` did not have this guard, meaning `import setup` has never worked before, so nothing will be broken by throwing an exception in that case.

> and we shouldn't counteract it.

Yes, we should counteract ways to call our code that have no use and will only happen by accident.